datafusion/physical-plan/src/joins/utils.rs

```diff
@@ -151,6 +151,82 @@ pub trait JoinHashMapType {
     fn get_map(&self) -> &RawTable<(u64, u64)>;
     /// Returns a reference to the next.
     fn get_list(&self) -> &Self::NextType;
+
```

Comment:
`JoinHashMap` and this trait are getting big enough maybe to warrant moving to its own module (in a separate PR, perhaps)

Reply:
Maybe partial join output will add some more logic to JoinHashMap and make separate module reasonable at that stage